.ci/pipeline/build-and-test-lnx.yml

```diff
@@ -47,7 +47,6 @@ steps:
       bash .ci/scripts/setup_sklearn.sh $(SKLEARN_VERSION)
       pip install --upgrade -r requirements-test.txt -r requirements-test-optional.txt
       pip install $(python .ci/scripts/get_compatible_scipy_version.py)
-      if [ $(echo $(PYTHON_VERSION) | grep '3.8\|3.9\|3.10') ]; then conda install -q -y -c intel dpnp; fi

```

Comment:
let's leave dpnp installation for testing. just specify version dpnp>=0.12*

Reply:
dpnp installation in testing stage will mix conda and pip commands which can result in env consistency problems in future. I prefer to preserve next order: conda env creation -> conda packages installation -> pypi packages installation

Reply:
On main branch we already installing `dpnp` on testing step, and everything works. Here we can just specify dpnp version: `dpnp>=0.12*`

I propose to solve future problems as they come.

In addition, I vote for the purity of the flow. We are building sklearnex without dpnp and therefore I see no reason to add it to the build env.

Reply:
dpnp is not essential scope of this PR so I removed it and we will discuss it later